### DISCUSSION TITLE
Maven project arguments for debug not working

### DISCUSSION THREAD

**Discussion by rhinoarmy** [apache/netbeans#6153] [Q&A]
I have a maven project, and it runs (bug green play button) like a champ out of the box, but debug is broken.

The way I did so was to supply a exec-maven-plugin and add the default arguments there in the pom.xml.
`exec-maven-plugin:plugin configuration > arguments > argument`
I then tried to debug (little green play with debug button) and thought this would work out of the box.
The arguments that were being passed in run, are not being passed when trying to debug.

Is there a reason these buttons should behave differently?
What is the best practice to get this working **without** using netbeans specific files (**pom.xml only**)?

**Comment by asbachb** [comment]
@musheno If this is still an open question: Can you share a minimal `pom.xml` showing how you configure the plugin?

**Comment by neilcsmith-net** [comment]
> What is the best practice to get this working without using netbeans specific files (pom.xml only)?

That feels like an oxymoron to me!  Set up the actions to run what you want.

Looking at the default action configuration, you might be able to set the `exec.appArgs` property in the pom.

**Reply by RogueZamboni** [reply]
So far as I can tell, this just doesn't work, netbeans somehow trashes whatever Maven arguments you try to use.  It's 2026 now, will likely never be fixed.
